src/http_api/service.rs

```diff
@@ -208,6 +208,30 @@ impl ReserveAuditorHttpService {
             config: self.gnosis_safe_config.clone(),
         })
     }
+
+    pub fn get_mint_info_by_block(&self, block_index: u64) -> Result<MintInfoResponse, Error> {
+        let conn = self.reserve_auditor_db.get_conn()?;
+        let mint_txs = MintTx::get_mint_txs_by_block_index(block_index, &conn)?;
+        let mint_config_txs = MintConfigTx::get_by_block_index(block_index, &conn)?;
+
+        let mut mints_with_configs = vec![];
+        for mint in mint_txs.iter() {
+            // In reality we should always have an id since this was returned from the database.
+            if let Some(id) = mint.id() {
+                if let Some(mint_config) = MintConfig::get_by_id(id, &conn)? {
+                    mints_with_configs.push(MintWithConfig {
+                        mint_tx: mint.clone(),
```

Comment:
Nit: I think this lets you avoid a clone

```suggestion
        for mint_tx in mint_txs.into_iter() {
            // In reality we should always have an id since this was returned from the database.
            if let Some(id) = mint_tx.id() {
                if let Some(mint_config) = MintConfig::get_by_id(id, &conn)? {
                    mints_with_configs.push(MintWithConfig {
                        mint_tx,
```